R/species.R

```diff
@@ -0,0 +1,87 @@
+#' Infer Species
+#'
+#' @param sset a \code{SigSet}
+#' @param df_as a data.frame of alignment score for each probe.
+#' @param topN Top n positive and negative probes used to infer species.
+#' @param threshold.pos pvalue < threshold.pos are considered to be positive probes (default is 0.01).
+#' @param threshold.neg pvalue > threshold.neg are considered to be negative probes (default is 0.2).
+#' @param ret.max whether to return the species with maximal AUC.
+#' @param balance whether to banlance the postive and negative probes size (default is TRUE).
+#' @return a list of auc, pvalue, species (NCBI official species names) and taxid.
+#' We infer species based on probes pvalues and alignment score.
+#' AUC was calculated for each specie, y_true is 1 or 0 
+#' for pval < threshold.pos or pval > threshold.neg, respeceively,
+#'
+#' Our function works on a single sample.
+#' @examples
+#' sset <- sesameDataGet('EPIC.1.LNCaP')$sset
+#' df_as <- sesameDataGet('df_as')
+#' res=inferSpecies(sset,df_as)
+#' res=inferSpecies(sset,df_as,ret.max=F)
+#' result=do.call(cbind, mclapply(idats, function(x) {
+#' 		sset=readIDATpair(x,manifest=mft, platform='mammal')
+#'      	res=inferSpecies(sset,df_as,ret.max=T)
+#'    },mc.cores=8)
+#'    )
+#' @export
+
+inferSpecies <- function(sset,df_as=NULL,topN=3000,
+			threshold.pos=0.01,threshold.neg=0.2,ret.max=T,
+			balance=T) {
+    if (is.null(df_as)) {
+    	df_as=sesameDataGet('df_as')
```

Comment:
I changed df_as to Mammal40.alignmentScore. Please test using sesameDataGet("Mammal40.alignmentScore") and update accordingly